lua/autorun/client/thirdperson.lua

```diff
@@ -21,28 +21,194 @@ hook.Add("Think", "ThirdPersonToggler", function()
 end)
 
 function UseThirdperson()
+    local wep = LocalPlayer():GetActiveWeapon()
     local wc = IsValid(LocalPlayer()) and IsValid(LocalPlayer():GetActiveWeapon()) and LocalPlayer():GetActiveWeapon():GetClass()
+    if LocalPlayer():IsPlayingTaunt() then return true end
+
+    if (wep.IsScoped and wep:IsScoped()) then
+        Thirdperson_Lerp = 0
+        Thirdperson_LerpSide = 0
+
+        return false
+    end
 
     return THIRDPERSON or wc == "weapon_fists" or LocalPlayer():HasWeapon("weapon_goohulk") or (wc == "weapon_garfield" and IsValid(LocalPlayer()) and not LocalPlayer():InVehicle())
 end
 
+local cwep = {
+weapon_pistol = true,
+weapon_357 = true,
+weapon_smg1 = true,
+weapon_ar2 = true,
+weapon_shotgun = true,
+weapon_crossbow = true,
+weapon_rpg = true,
+weapon_grenade = true,
+}
+
+-- taunts where the feet are used
+local LockTaunts = {
+	ACT_GMOD_TAUNT_MUSCLE = true,
+	ACT_GMOD_TAUNT_DANCE = true,
+	ACT_GMOD_TAUNT_ROBOT = true,
+	ACT_GMOD_TAUNT_PERSISTENCE = true,
+	ACT_GMOD_TAUNT_LAUGH = true,
+	ACT_GMOD_TAUNT_CHEER = true
+}
+
+
+function IsLockedByTaunting(ply)
+    if ply:IsPlayingTaunt() then
```

Comment:
`return ply:IsPlayingTaunt() and LockTaunts[ply:GetSequenceActivityName(ply:GetLayerSequence(5))]`